nistats/first_level_model.py

```diff
@@ -0,0 +1,561 @@
+        for val, result in zip(vals, ar_result):
+            results[val] = result
+        del vals
+        del ar_result
+
+    else:
+        labels = np.zeros(Y.shape[1])
+        results = {0.0: ols_result}
+
+    return labels, results
+
+
+class FirstLevelModel(BaseEstimator, TransformerMixin, CacheMixin):
+    """ Implementation of the General Linear Model for single session fMRI data
+
+    Parameters
+    ----------
+
+    t_r: float
+        This parameter indicates repetition times of the experimental runs.
+        It is necessary to correctly consider times in the design matrix.
+        This parameter is also passed to nilearn.signal.clean.
+        Please see the related documentation for details.
+
+    slice_time_ref: float, optional (default 0.)
+        This parameter indicates the time of the reference slice used in the
+        slice timing preprocessing step of the experimental runs. It is
+        expressed as a percentage of the t_r (time repetition), so it can have
+        values between 0. and 1.
+
+    hrf_model : string, optional
+        This parameter specifies the hemodynamic response function (HRF) for
+        the design matrices. It can be 'canonical', 'canonical with derivative'
+        or 'fir'.
+
+    drift_model : string, optional
+        This parameter specifies the desired drift model for the design
+        matrices. It can be 'polynomial', 'cosine' or 'blank'.
+
+    period_cut : float, optional
+        This parameter specifies the cut period of the low-pass filter in
+        seconds for the design matrices.
+
+    drift_order : int, optional
+        This parameter specifices the order of the drift model (in case it is
+        polynomial) for the design matrices.
+
+    fir_delays : array of shape(n_onsets) or list, optional
+        In case of FIR design, yields the array of delays used in the FIR
+        model.
+
+    min_onset : float, optional
+        This parameter specifies the minimal onset relative to the design
+        (in seconds). Events that start before (slice_time_ref * t_r +
+        min_onset) are not considered.
+
+    mask: Niimg-like, NiftiMasker or MultiNiftiMasker object, optional,
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask will be used. If no mask is given,
+        it will be computed automatically by a MultiNiftiMasker with default
+        parameters.
+
+    target_affine: 3x3 or 4x4 matrix, optional
+        This parameter is passed to nilearn.image.resample_img. Please see the
+        related documentation for details.
+
+    target_shape: 3-tuple of integers, optional
+        This parameter is passed to nilearn.image.resample_img. Please see the
+        related documentation for details.
+
+    low_pass: False or float, optional
+        This parameter is passed to nilearn.signal.clean.
+        Please see the related documentation for details.
+
+    high_pass: False or float, optional
```

Comment:
parameters clash: we now also have a period_cut parameter that does the same thing.
I think that the masker should be called with low_pass/high_pass parameters set to False automatically, so that all the filtering is explicit in the design matrix. the low_pass and high_pass parameters would then disappear from this class constructor.